Data race flake in TestGetAPISpecsRPCSuccess

```
time="Aug  1 13:55:17" level=info msg="Starting Poller" 
time="Aug  1 13:55:17" level=error msg="Failed to parse URL! parse : empty url" 
time="Aug  1 13:55:17" level=error msg="Init failed for this webhook, invalid URL, URL must be absolute" 
time="Aug  1 13:55:17" level=info msg="Setting up analytics normaliser" 
time="Aug  1 13:55:17" level=info msg="PIDFile location set to: /var/run/tyk-gateway.pid" 
time="Aug  1 13:55:17" level=error msg="Failed to write PIDFile: error opening pidfile /var/run/tyk-gateway.pid: open /var/run/tyk-gateway.pid354010492: permission denied" 
time="Aug  1 13:55:17" level=error msg="Failed ot get host pid: open /var/run/tyk-gateway.pid: no such file or directory" 
time="Aug  1 13:55:17" level=error msg="Instrumentation is enabled, but no connectionstring set for statsd" 
time="Aug  1 13:55:17" level=info msg="Initialising Tyk REST API Endpoints" 
time="Aug  1 13:55:17" level=warning msg="Legacy path detected! Upgrade to extended." 
time="Aug  1 13:55:17" level=warning msg="Legacy path detected! Upgrade to extended." 
time="Aug  1 13:55:17" level=warning msg="Legacy path detected! Upgrade to extended." 
time="Aug  1 13:55:17" level=warning msg="Legacy path detected! Upgrade to extended." 
time="Aug  1 13:55:17" level=warning msg="Legacy path detected! Upgrade to extended." 
time="Aug  1 13:55:17" level=warning msg="Legacy path detected! Upgrade to extended." 
time="Aug  1 13:55:17" level=warning msg="Legacy path detected! Upgrade to extended." 
time="Aug  1 13:55:17" level=warning msg="Legacy path detected! Upgrade to extended." 
time="Aug  1 13:55:17" level=warning msg="Legacy path detected! Upgrade to extended." 
time="Aug  1 13:55:17" level=warning msg="Legacy path detected! Upgrade to extended." 
time="Aug  1 13:55:17" level=info msg="Setting new RPC connection!" 
time="Aug  1 13:55:17" level=info msg="Storing RPC backup" 
time="Aug  1 13:55:17" level=info msg="--> Connecting to DB" 
time="Aug  1 13:55:17" level=info msg="--> Connected to DB" 
time="Aug  1 13:55:17" level=error msg="Failed decode: invalid character 'm' looking for beginning of value" 
time="Aug  1 13:55:17" level=info msg="Detected 0 APIs" 
time="Aug  1 13:55:17" level=info msg="Setting new RPC connection!" 
==================
WARNING: DATA RACE
Read at 0x0000018a8fa0 by goroutine 361:
  github.com/TykTechnologies/tyk.(*RPCStorageHandler).Connect.func1()
      /home/travis/gopath/src/github.com/TykTechnologies/tyk/rpc_storage_handler.go:188 +0x2cc
  github.com/TykTechnologies/tyk/vendor/github.com/lonelycode/gorpc.clientHandler.func1()
      /home/travis/gopath/src/github.com/TykTechnologies/tyk/vendor/github.com/lonelycode/gorpc/client.go:510 +0x6f

Previous write at 0x0000018a8fa0 by goroutine 328:
  github.com/TykTechnologies/tyk.(*RPCStorageHandler).Connect()
      /home/travis/gopath/src/github.com/TykTechnologies/tyk/rpc_storage_handler.go:144 +0x278
  github.com/TykTechnologies/tyk.APIDefinitionLoader.FromRPC()
      /home/travis/gopath/src/github.com/TykTechnologies/tyk/api_definition.go:311 +0x130
  github.com/TykTechnologies/tyk.getAPISpecs()
      /home/travis/gopath/src/github.com/TykTechnologies/tyk/main.go:227 +0xaa0
  github.com/TykTechnologies/tyk.TestGetAPISpecsRPCSuccess()
      /home/travis/gopath/src/github.com/TykTechnologies/tyk/api_definition_test.go:376 +0x148
  testing.tRunner()
      /home/travis/.gimme/versions/go1.8.3.linux.amd64/src/testing/testing.go:657 +0x107

Goroutine 361 (running) created at:
  github.com/TykTechnologies/tyk/vendor/github.com/lonelycode/gorpc.clientHandler()
      /home/travis/gopath/src/github.com/TykTechnologies/tyk/vendor/github.com/lonelycode/gorpc/client.go:515 +0x14a

Goroutine 328 (running) created at:
  testing.(*T).Run()
      /home/travis/.gimme/versions/go1.8.3.linux.amd64/src/testing/testing.go:697 +0x543
  testing.runTests.func1()
      /home/travis/.gimme/versions/go1.8.3.linux.amd64/src/testing/testing.go:882 +0xaa
  testing.tRunner()
      /home/travis/.gimme/versions/go1.8.3.linux.amd64/src/testing/testing.go:657 +0x107
  testing.runTests()
      /home/travis/.gimme/versions/go1.8.3.linux.amd64/src/testing/testing.go:888 +0x4e0
  testing.(*M).Run()
      /home/travis/.gimme/versions/go1.8.3.linux.amd64/src/testing/testing.go:822 +0x1c3
  github.com/TykTechnologies/tyk.TestMain()
      /home/travis/gopath/src/github.com/TykTechnologies/tyk/gateway_test.go:178 +0x464
  main.main()
      github.com/TykTechnologies/tyk/_test/_testmain.go:304 +0x20f
==================
time="Aug  1 13:55:18" level=info msg="Storing RPC backup" 
time="Aug  1 13:55:18" level=info msg="--> Connecting to DB" 
time="Aug  1 13:55:18" level=info msg="--> Connected to DB" 
time="Aug  1 13:55:18" level=info msg="Detected 1 APIs" 
--- FAIL: TestGetAPISpecsRPCSuccess (0.32s)
	testing.go:610: race detected during execution of test
```

--- a/rpc_storage_handler.go
+++ b/rpc_storage_handler.go
@@ -125,7 +125,6 @@ func (r *RPCStorageHandler) ReConnect() {
 
 var RPCCLientSingleton *gorpc.Client
 var RPCCLientSingletonMu sync.Mutex
-var RPCClientSingletonConnectionID string
 var RPCFuncClientSingleton *gorpc.DispatcherClient
 var RPCGlobalCache = cache.New(30*time.Second, 15*time.Second)
 var RPCClientIsConnected bool
@@ -141,11 +140,11 @@ func (r *RPCStorageHandler) Connect() bool {
 	// Set up the cache
 	log.Info("Setting new RPC connection!")
 
-	RPCClientSingletonConnectionID = uuid.NewV4().String()
+	connID := uuid.NewV4().String()
 
 	// Length should fit into 1 byte. Protection if we decide change uuid in future.
-	if len(RPCClientSingletonConnectionID) > 255 {
-		panic("RPCClientSingletonConnectionID is too long")
+	if len(connID) > 255 {
+		panic("connID is too long")
 	}
 
 	if globalConf.SlaveOptions.UseSSL {
@@ -185,8 +184,8 @@ func (r *RPCStorageHandler) Connect() bool {
 		}
 
 		conn.Write([]byte("proto2"))
-		conn.Write([]byte{byte(len(RPCClientSingletonConnectionID))})
-		conn.Write([]byte(RPCClientSingletonConnectionID))
+		conn.Write([]byte{byte(len(connID))})
+		conn.Write([]byte(connID))
 		return conn, nil
 	}
 	RPCCLientSingleton.Start()

